### ISSUE TITLE
.desktop file Exec line

### ISSUE THREAD

**Issue by ghost** [WHO53/wvkbd-phosh-osk#1]
I thought I posted this a day or two ago, but apparently I was in the wrong tab and actually posted it in agx/phosh-osk-stub. 🤦

Anyway, I'm trying to get this setup on PostmarketOS, and used dpkg to install the .deb file, but while it did appear to install, it failed to change the Exec line (I feel like `update-alternatives` is a Debian-specific thing?). Also, weirdly, while both wvkbd-mobintl, and wvkbd-dbus both appear in /usr/bin, trying to run either of them gives me "not found"?

So basically, my questions are 1. which of these executables do I put in the Exec line of the desktop file?, or 2. if neither, what *do* I put in the desktop file? I did try wvkbd-dbus, and that just gave me an error when Phosh started up.

**Comment by who53** [comment]
https://github.com/WHO53/wvkbd-phosh-osk/blob/bfb884d4821c7fde9b4ef5d0bb4556eb2437d7a5/README.md

No. 4

**Comment by ghost** [comment]
Oh, wow; how incredibly observant of me.

The funny part is I definitely saw that before installation, but came back [here](https://github.com/WHO53/wvkbd-phosh-osk/blob/master/README.md) afterward, and I guess I assumed I'd imagined it?

Anyway, thanks. Much appreciated!

**Comment by ghost** [comment]
Okay, just rebooted, and while I didn't get the error I got previously, I don't get the keyboard either. Long-pressing the bar at the bottom of the screen just causes it to jiggle, but I'll do some digging and see what I can figure out on my own. This feels like progress though, at least.

**Comment by who53** [comment]
Check if both processes are running, wvkbd-dbus and wvkbd-mobintl.

You can enable debug for wvkbd-dbus here. (I don't think something wrong here but you can check)
https://github.com/WHO53/wvkbd-phosh-osk/blob/101155333bc5c6c6cbfdb549f2e99d50a12c18c2/wvkbd-dbus.c#L55

while I think it should just work
Most probably one of them is not running 

Also if wvkbd-mobintl two processes are running kill both of them, start a new one

**Comment by ghost** [comment]
Neither are running. I ran `file` and `ldd` on both, and got the following for wvkbd-dbus:
```
oneplus-fajita:/usr/bin$ ldd wvkbd-dbus
	/lib/ld-linux-aarch64.so.1 (0xffffbc3e5000)
	libwayland-client.so.0 => /usr/lib/libwayland-client.so.0 (0xffffbc3a3000)
	libgio-2.0.so.0 => /usr/lib/libgio-2.0.so.0 (0xffffbc000000)
	libgobject-2.0.so.0 => /usr/lib/libgobject-2.0.so.0 (0xffffbc321000)
	libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0xffffbbe9e000)
	libc.so.6 => /lib/ld-linux-aarch64.so.1 (0xffffbc3e5000)
Error loading shared library ld-linux-aarch64.so.1: No such file or directory (needed by wvkbd-dbus)
	libffi.so.8 => /usr/lib/libffi.so.8 (0xffffbc300000)
	libgmodule-2.0.so.0 => /usr/lib/libgmodule-2.0.so.0 (0xffffbc2df000)
	libz.so.1 => /lib/libz.so.1 (0xffffbc2ae000)
	libmount.so.1 => /lib/libmount.so.1 (0xffffbc24d000)
	libintl.so.8 => /usr/lib/libintl.so.8 (0xffffbc22b000)
	libpcre2-8.so.0 => /usr/lib/libpcre2-8.so.0 (0xffffbbded000)
	libblkid.so.1 => /lib/libblkid.so.1 (0xffffbbdac000)
	libeconf.so.0 => /usr/lib/libeconf.so.0 (0xffffbc209000)
Error relocating wvkbd-dbus: __snprintf_chk: symbol not found
```

So I suspect these missing shared libraries are probably the culprit.

**Comment by ghost** [comment]
Woo! All it took was `apk add gcompat` and a reboot, and there it is! Hopefully less crashy than when I installed the original wvkbd before I knew about all the dbus stuff :p
